firestore/src/android/field_value_android.cc

```diff
@@ -98,39 +99,50 @@ FieldValue FieldValueInternal::Create(Env& env,
 FieldValueInternal::FieldValueInternal() : cached_type_(Type::kNull) {}
 
 FieldValueInternal::FieldValueInternal(const Object& object)
-    : object_(object), cached_type_(Type::kNull) {}
+    : cached_type_(Type::kNull) {
+  Env env;
+  object_ = ArenaRef(env, object);
+}
 
 FieldValueInternal::FieldValueInternal(Type type, const Object& object)
-    : object_(object), cached_type_(type) {}
+    : cached_type_(type) {
+  Env env;
+  object_ = ArenaRef(env, object);
+}
 
 FieldValueInternal::FieldValueInternal(bool value)
     : cached_type_(Type::kBoolean) {
-  Env env = GetEnv();
-  object_ = Boolean::Create(env, value);
+  Env env;
+  auto boxed_ptr = Boolean::Create(env, value);
```

Comment:
nit: Remove the `boxed_ptr` local variable and just specify `Boolean::Create(env, value)` as an argument to the ArenaRef constructor.

IMO the `boxed_ptr` local variable just adds noise and detracts from readability. Please make this change elsewhere in this file too.